server/text_generation_server/models/flash_causal_lm.py

```diff
@@ -713,6 +715,37 @@ def __init__(
             sliding_window=sliding_window,
         )
 
+        if kv_cache_dtype == "fp8":
+            self.kv_cache_dtype = torch.uint8
+        else:
+            self.kv_cache_dtype = self.dtype
+
+        if kv_cache_dtype == "fp8" and SYSTEM == "rocm":
+            logger.info(f"Using KV cache data type: {kv_cache_dtype}")
+            # Currently scaled KV cache is only enabled on ROCm
+            if quantization_param_path is not None:
+                if callable(getattr(self.model, "load_kv_cache_scales", None)):
+                    self.model.load_kv_cache_scales(quantization_param_path)
```

Comment:
If done correctly, you don't need any of that because you handle this at load time (and every model will need to be updated to support the new values)